### ISSUE TITLE
Max FPS and Alarm Max FPS disable when Source Type IP Camera

### ISSUE THREAD

**Issue by SteveGilvarry** [ZoneMinder/ZoneMinder#754]
Mentioned on irc that we should fix this so logging the issue so we remember to get to it. We should be able to manage in the GUI to clear the values and hide the fields when FFMPEG, Remote, cURL and Libvlc Source Types are selected. I assume these sources are only used for IP Cameras?

The other option is to leave the fields as is but make sure the values have no impact, but that could still confuse users.

**Comment by knnniggett** [comment]
Agreed.
What complicates this is that there is at least one case I can think of where we have a source type of "remote" and we need those fields to exists.
(typical older) IP cameras that create snapshots, rather than send an mjpeg or h264 stream will need those fields.

**Comment by knnniggett** [comment]
I just thought about another potential case....
If an ffmpeg camera goes offline, the fps spikes which pegs the cpu.  A possible workaround is to place a value in the MAX Fps field higher than the normal frame rate.  Of course, this is just a workaround so one could argue that we should address the root cause instead.  

Perhaps we could make the fields "advanced", requiring a check box to be checked in order to make them re-appear.

**Comment by newburns** [comment]
I was going to put in my 2 cents, but Andrew beat me to it.
I put in the Max FPS so that my CPU doesn't go to 100 load.

**Comment by SteveGilvarry** [comment]
Yes I wanted to find the root cause, as I move my dev laptop between home and office and have different cameras so if I forget to disable and enable then things go nuts and I spend 20minutes waiting for responses to shut down. I am all for it reconnecting, but it needs a throttle on it. Do we have an issue for that also?
Back to the jpeg case I will have to see if I can find some examples of how people use them to see if we can find a combination that says enable Max FPS or disable.

**Comment by SteveGilvarry** [comment]
Related https://github.com/ZoneMinder/ZoneMinder/issues/365

**Comment by connortechnology** [comment]
I think maybe they should be relevant.  

For example ffmpeg... maybe we don't need to do motion detection on every frame.  Maybe every 5th.. even though you record all.  

I agree that they should only be available if they have an effect.

**Comment by arjunroychowdhury** [comment]
Hello, I am curious why the CPU spikes if the camera is offline? I had kept my FPS in ZM blank, but given that it does spike, I've kept my ZM FPS MAX at a value higher than the in-camera FPS. But it wasn't evident to me why this is happening? I took at look at zm_ffmpeg_camera.cpp --> seems if it fails to open an ffmpeg stream, it waits for the configured ffmpeg timeout and launches another thread to try again - I don't see max-fps being used here. So I'm curious how this helps (I confirmed, it does help)
